File: web/tawagoto/tawagoto/v2/model/Product.php
<?php

  class Product{
    public $id;
    //public Category $cat; à voir plus tard
    public $cat;
    public $subcat;
    public $name;
    public $price;
    public $description;
    public $alertThreshold;
    public $stock;
    
    
    
    function __construct($id, $cat, $subcat,
        $name, $price, $description, $alertThreshold,
        $stock){
            $this->id = $id;
            $this->cat = $cat;
            $this->subcat = $subcat;
            $this->name = $name;
            $this->price = $price;
            $this->description = $description;
            $this->alertThreshold = $alertThreshold;
            $this->stock = $stock;
            
    }
    
    function __toString(){
        return $this->id . "</br>" .
            $this->cat . "</br>" .
            $this->subcat . "</br>" .
            $this->name . "</br>" .
            $this->price . "</br>" .
            $this->description . "</br>" .
            $this->alertThreshold . "</br>" .
            $this->stock . "</br>";
    }
    
    function updateStock(){
        
    }
    //static function updateStock($prod){} //
    
    
    
    //getters
    public function getId()
    {
        return $this->id;
    }

    public function getCat()
    {
        return $this->cat;
    }

    public function getSubcat()
    {
        return $this->subcat;
    }

    public function getName()
    {
        return $this->name;
    }

    public function getPrice()
    {
        return $this->price;
    }

    public function getDescription()
    {
        return $this->description;
    }

    public function getAlertThreshold()
    {
        return $this->alertThreshold;
    }

    public function getStock()
    {
        return $this->stock;
    }

    //setters
    public function setId($id)
    {
        $this->id = $id;
    }

    public function setCat($cat)
    {
        $this->cat = $cat;
    }

    public function setSubcat($subcat)
    {
        $this->subcat = $subcat;
    }

    public function setName($name)
    {
        $this->name = $name;
    }

    public function setPrice($price)
    {
        $this->price = $price;
    }

    public function setDescription($description)
    {
        $this->description = $description;
    }

    public function setAlertThreshold($alertThreshold)
    {
        $this->alertThreshold = $alertThreshold;
    }

    public function setStock($stock)
    {
        $this->stock = $stock;
    }

    
    //fonction de test
    
    public function getPathImage(){
        return "https://th.bing.com/th/id/OIP.h0-OFPdOWon5agPhwfexWQHaEL?w=300&h=180&c=7&r=0&o=5&dpr=1.5&pid=1.7";
    }
    
    public function getScore(){
        return 5;
    }

  }

File: Lot2/Livrable 3/Appli Java (code source)/eCommerce/src/main/java/model/Product.java
package model;

import java.util.ArrayList;

public class Product {
	private String name, description;
	private Category cat;
	private Subcategory subcat;
	private double price;
	private int ID, alertThreshold, stock;
	private Promotion promotion;
	
	public static ArrayList<Product> buffer = new ArrayList<Product>();

	public Product(int ID, Category cat, Subcategory subcat, String name, double price, String description,
			int alertThreshold, int stock, Promotion promotion) {

		this.ID = ID;
		this.name = name;
		this.description = description;
		this.cat = cat;
		this.subcat = subcat;
		this.price = price;
		this.promotion = promotion;
		this.alertThreshold = alertThreshold;
		this.stock = stock;
	}
	
	public Product(Category cat, Subcategory subcat, String name, double price, String description,
			int alertThreshold, int stock, Promotion promotion) {
		this(-1, cat, subcat, name, price, description, alertThreshold, stock, promotion);
	}
	
	public static void addAll(ArrayList<Product> arrayList) {
		for(int i = 0; i < arrayList.size(); i++) {
			buffer.add(arrayList.get(i));
		}
	}
	
	public static void overwriteBuffer(ArrayList<Product> arrayList) {
		buffer.clear();
		addAll(arrayList);
	}
	
	public static Product getProductByID(int ID) {
		Product product, result = null;
		for(int i = 0; i < buffer.size() && result == null; i++) {
			product = buffer.get(i);
			if(product.getID() == ID) {
				result = product;
			}
		}
		
		return result;
	}
	
	public String toString() { return this.name ; }
	public String testField() { return name + "\n" + cat.toString() + "\n" + price + "\n" + stock; }
 
	public int getID() { return ID ; }
	public String getName() { return name ; }
	public String getDescription() { return description ; }
	public Category getCat() { return cat ; }
	public Subcategory getSubcat() { return subcat ; }
	public double getPrice() { return price ; }
	public Promotion getPromotion() { return promotion ; }
	public int getAlertThreshold() { return alertThreshold ; }
	public int getStock() { return stock ; }

	public void setName(String name) { this.name = name ; }
	public void setDescription(String description) { this.description = description ; }
	public void setCat(Category cat) { this.cat = cat ; }
	public void setSubcat(Subcategory subcat) { this.subcat = subcat ; }
	public void setPrice(double price) { this.price = price ; }
	public void setPromotion(Promotion promotion) { this.promotion = promotion ; }
	public void setAlertThreshold(int alertThreshold) { this.alertThreshold = alertThreshold ; }
	public void setStock(int stock) { this.stock = stock ; }
}